src/main/java/oshi/software/os/mac/MacOperatingSystem.java

```diff
@@ -26,4 +27,14 @@ public MacOperatingSystem() {
         this.family = System.getProperty("os.name");
         this.version = new MacOSVersionInfoEx();
     }
+
+    @Override
+    public long getOpenDescriptors() {
+        return SysctlUtil.sysctl("kern.num_files", -1);
```

Comment:
Need the default to be -1L to use the overloaded method that returns a long, otherwise you're returning an int and casting to long.